qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCodeActions.java

```diff
@@ -186,15 +199,52 @@ private static void doCodeActionToDisableValidation(Template template, List<Diag
 		codeActions.add(disableValidationForTemplateQuickFix);
 	}
 
+	/**
+	 * Create CodeAction for deprecated `with` Qute syntax.
+	 *
+	 * @param template    the Qute template.
+	 * @param diagnostic  the diagnostic list that this CodeAction will fix.
+	 * @param codeActions the list of CodeActions to perform.
+	 * @throws BadLocationException
+	 *
+	 */
+	private void doCodeActionsForNotRecommendedWithSection(Template template, Diagnostic diagnostic,
+			List<CodeAction> codeActions) {
+		Range withSectionRange = diagnostic.getRange();
+		try {
+			int withSectionStart = template.offsetAt(withSectionRange.getStart());
+			int withSectionEnd = template.offsetAt(withSectionRange.getEnd());
+			WithSection withSection = (WithSection) template.findNodeAt(withSectionStart);
+			String javaResourceClassName = withSection.getObjectParameter().getName();
+			List<Node> withExpressions = withSection.getChildren().stream().filter(s -> s instanceof Expression)
+					.collect(Collectors.toList());
+			List<String> letExpressionParameterList = new ArrayList<String>();
+			for (Node expression : withExpressions) {
+				letExpressionParameterList.add(MessageFormat.format("{1}={0}.{1}", javaResourceClassName,
+						((Expression) expression).getContent()));
+			}
+			String letExpressionParameters = String.join(" ", letExpressionParameterList);
+			String withSectionText = template.getText(withSectionStart, withSectionEnd);

```

Comment:
I think it should be cleaner to create a textedit per update. Don't replace the full text of #with but create several text edit per update:

 * #with -> #let
 * add parameters in the let
 * /with -> /let

Reply:
Now that I have access to the WithSection, this should be easier to do